mappers/jsonProvider/sources/googleTakeout.js

```diff
@@ -0,0 +1,57 @@
+const { difference, equals, replace } = require('ramda')
+const { getUnixTime } = require('date-fns')
+
+const getExpectedInputKeys = objectTypeId => {
+  switch (objectTypeId) {
+    // Watch History
+    case 4311: {
+      return ['header', 'title', 'titleUrl', 'subtitles', 'time', 'products']
+    }
+    // Search History
+    case 4312: {
+      return ['header', 'title', 'titleUrl', 'time', 'products']
+    }
+  }
+}
+
+const isInvalid = (keys, objectTypeId) => {
+  const expectedInputKeys = getExpectedInputKeys(objectTypeId)
+
+  if (!equals(expectedInputKeys, keys)) {
```

Comment:
maybe such strict enforcement is not needed here - I'd just check if we have enough inside, and not fail if there are extra stuff

not sure if order affects `equals`, but we should also never rely on order of keys in an object, because it can change randomly